src/components/ButtonGroup/ButtonGroup.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { arrayOf, bool, node } from 'prop-types';
+import glamorous from 'glamorous';
+
+import Button from '../Button/Button';
+import constants from '../../constants';
+
+const { borderRadius } = constants;
+const { Div } = glamorous;
+
+const ButtonGroupItem = glamorous(Button)({
+   borderRadius: 0,
+   borderRightWidth: 0,
+
+   '&:first-of-type': {
+      borderTopLeftRadius: borderRadius,
+      borderBottomLeftRadius: borderRadius,
+   },
+   '&:last-of-type': {
+      borderTopRightRadius: borderRadius,
+      borderBottomRightRadius: borderRadius,
+      borderRightWidth: 1,
+   },
+});
+
+const ButtonGroup = ({ children, disabled, ...props }) => (
+   <Div display="inline-block" {...props}>
+      {React.Children.map(children, child => {
+         if (child.type !== Button) {
+            return child;
+         }
+         return <ButtonGroupItem {...child.props} disabled={disabled} />;
+      })}
```

Comment:
Doesn't this mean that you can only ever disable the entire button group? What if we want to disable just a single button in the group?

I think you could do something like:
`return <ButtonGroupItem {...child.props} disabled={disabled || child.props.disabled} />;`

Reply:
Good catch I'll fix that right away.